File: src/service/service.rb
class Service

  attr_reader :name
  attr_reader :url
  attr_reader :category
  attr_reader :icon
  attr_reader :image_url
  attr_reader :opencontainers_image_title
  attr_reader :hostname
  attr_reader :ignore
  alias :ignore? :ignore

  def initialize(name, url, category, icon, image_url, opencontainers_image_title, hostname, ignore=false)
    @name = name
    @url = url
    @category = category
    @icon = icon
    @image_url = image_url
    @opencontainers_image_title = opencontainers_image_title
    @hostname = hostname
    @ignore = false
  end

  def as_json(options={})
    {
      name: @name,
      url: @url,
      category: @category,
      icon: @icon,
      image_url: @image_url,
      opencontainers_image_title: @opencontainers_image_title,
      hostname: @hostname
    }
  end

  def to_json(*options)
    as_json(*options).to_json(*options)
  end

end

File: dotnet/Dcm.Shared/Models/Service.cs
namespace Dcm.Shared.Models;
public class Service
{
    public string Name { get; set; }
    public string Url { get; set; }
    public string Category { get; set; }
    public string Icon { get; set; }
    public string ImageUrl { get; set; }
    public string Hostname { get; set; }
    public bool Ignore { get; set; }

    public Service()
    {
    }

    public Service(string name, string url, string category, string icon, string imageUrl, 
                   string hostname, bool ignore)
    {
      Name = name;
      Url = url;
      Category = category;
      Icon = icon;
      ImageUrl = imageUrl;
      Hostname = hostname;
      Ignore = ignore;
    }

    public override string ToString()
    {
        return $"{nameof(Name)}: {Name}, {nameof(Url)}: {Url}, {nameof(Category)}: {Category}, {nameof(Icon)}: {Icon}, {nameof(ImageUrl)}: {ImageUrl}, {nameof(Hostname)}: {Hostname}, {nameof(Ignore)}: {Ignore}";
    }
}
